tests/test_rm.py

```diff
@@ -0,0 +1,215 @@
+import pytest
+from pypdf import PdfReader
+
+from .conftest import RESOURCES_ROOT, chdir, run_cli
+
+
+def test_rm_incorrect_number_of_args(capsys, tmp_path):
+    with chdir(tmp_path):
+        exit_code = run_cli(["rm", str(RESOURCES_ROOT / "box.pdf")])
+    assert exit_code == 2
+    captured = capsys.readouterr()
+    assert "Missing argument" in captured.err
+
+
+def test_rm_subset_ok(capsys, tmp_path):
+    with chdir(tmp_path):
+        exit_code = run_cli(
+            [
+                "rm",
+                str(RESOURCES_ROOT / "GeoBase_NHNC1_Data_Model_UML_EN.pdf"),
+                "13:15",
+                "--output",
+                "./out.pdf",
+            ]
+        )
+    captured = capsys.readouterr()
+    assert exit_code == 0, captured
+    assert not captured.err
+    inp_reader = PdfReader(
+        RESOURCES_ROOT / "GeoBase_NHNC1_Data_Model_UML_EN.pdf"
+    )
+    out_reader = PdfReader(tmp_path / "out.pdf")
+    assert len(out_reader.pages) == len(inp_reader.pages) - 2
+
+
+@pytest.mark.parametrize(
+    "page_range",
+    ["a", "-", "1-", "1-1-1", "1:1:1:1"],
+)
+def test_rm_subset_invalid_args(capsys, tmp_path, page_range):
+    with chdir(tmp_path):
+        exit_code = run_cli(
+            [
+                "rm",
+                str(RESOURCES_ROOT / "jpeg.pdf"),
+                page_range,
+                "--output",
+                "./out.pdf",
+            ]
+        )
+    captured = capsys.readouterr()
+    assert exit_code == 2, captured
+    assert "Invalid file path or page range provided" in captured.err
+
+
+@pytest.mark.skip(reason="This check is not implemented yet")
+def test_rm_subset_warn_on_missing_pages(capsys, tmp_path):
+    with chdir(tmp_path):
+        exit_code = run_cli(
+            [
+                "rm",
+                str(RESOURCES_ROOT / "jpeg.pdf"),
+                "2",
+                "--output",
+                "./out.pdf",
+            ]
+        )
+    captured = capsys.readouterr()
+    assert exit_code == 0, captured
+    assert "WARN" in captured.out
+
+
+@pytest.mark.xfail()  # There is currently a bug there
```

Comment:
Why is there a bug there?

Reply:
It was just a note to remember the functionality of the decorated, I forgot to delete it :)

Reply:
Alright for the comment, but why marking this unit test as failing?

Reply:
Indeed that was an error, I have just uploaded a new version of the test file with the bugs solved